3rdparty/snmalloc/CMakeLists.txt

```diff
@@ -1,7 +1,12 @@
 # Copyright (c) Open Enclave SDK contributors.
 # Licensed under the MIT License.
 
+check_submodule_not_empty("${PROJECT_SOURCE_DIR}/3rdparty/snmalloc/snmalloc")
 add_enclave_library(oesnmalloc OBJECT allocator.cpp)
+# Add explicit dependency to musl_includes and libcxx includes.
```

Comment:
If you add the following, then it should pull in the required compiler settings from snmalloc:
```Cmake
# Include snmalloc library
set(SNMALLOC_ONLY_HEADER_LIBRARY ON)
add_subdirectory(snmalloc EXCLUDE_FROM_ALL)
target_link_libraries(oesnmalloc PRIVATE snmalloc_lib)
```
This has the advantage that if we alter our dependencies then it should get updated here automatically.  However, as you are building in a constrained setting, we may specify things that don't work here.